File: src/main/java/insogroup/utils/utils/IOC/strategies/OneObjectIOCStrategy.java
package insogroup.utils.utils.IOC.strategies;

import insogroup.utils.utils.IOC.exceptions.ResolveStrategyException;
import insogroup.utils.utils.IOC.interfaces.IOCStrategy;

public class OneObjectIOCStrategy implements IOCStrategy {

    protected Object targetObject;

    public OneObjectIOCStrategy(Object targetObject) {
        this.targetObject = targetObject;
    }

    /**
     * @param args Args for using in strategy
     * @return Always return the object which was set in constructor
     * @throws ResolveStrategyException
     */
    @Override
    public Object getInstance(Object... args) throws ResolveStrategyException {
        return targetObject;
    }
}

File: src/main/kotlin/com/github/insanusmokrassar/utils/IOC/strategies/OneObjectIOCStrategy.kt
package com.github.insanusmokrassar.utils.IOC.strategies

import com.github.insanusmokrassar.utils.IOC.exceptions.ResolveStrategyException
import com.github.insanusmokrassar.utils.IOC.interfaces.IOCStrategy

class OneObjectIOCStrategy(protected var targetObject: Any) : IOCStrategy {

    /**
     * @param args Args for using in strategy
     * *
     * @return Always return the object which was set in constructor
     * *
     * @throws ResolveStrategyException
     */
    @Throws(ResolveStrategyException::class)
    override fun getInstance(vararg args: Any): Any {
        return targetObject
    }
}
